File: codingtest/pyrhon lv1~/동영상 재생기.py
from datetime import datetime, timedelta

def solution(video_len, pos, op_start, op_end, commands):
    conv_time = lambda target : datetime.strptime(target, "%M:%S")
    video_len = conv_time(video_len)
    pos = conv_time(pos)
    op_start = conv_time(op_start)
    op_end = conv_time(op_end)

    if op_start <= pos and pos <= op_end:
        pos = op_end

    for command in commands:
        if command == 'prev':
            if pos - timedelta(seconds=10) > conv_time("0:0"):
                pos = pos - timedelta(seconds=10)
            else:
                pos = conv_time("0:0")
        elif command == 'next':
            if pos + timedelta(seconds=10) < video_len:
                pos = pos + timedelta(seconds=10)
            else:
                pos = video_len

        if op_start <= pos and pos <= op_end:
            pos = op_end
    return pos.strftime("%M:%S")

File: codingtest/cpp lv1~3/lv1/동영상_재생기.cpp
#include<iostream>
#include<vector>
#include<string>
using namespace std;// 이름공간을 사용해야 string을 그냥 사용할 수 있다?

string solution(const string video_len, const string pos, const string op_start,
	const string op_end, vector<string>commands, size_t commands_len) {
	string answer;
	/*string의 형태로 분과 초를 주기 때문에 모두 초단위로 바꿔서 사용할 수 밖에 없다*/
	int Pos = stoi(pos.substr(0, 2)) * 60 + stoi(pos.substr(3, 2));
	int Op_start = stoi(video_len.substr(0, 2)) * 60 + stoi(video_len.substr(3, 2));
	int Op_end = stoi(op_end.substr(0, 2)) * 60 + stoi(op_end.substr(3, 2));
	int Video_len = stoi(video_len.substr(0, 2)) * 60 + stoi(video_len.substr(3, 2));

	if (Op_start <= Pos && Pos <= Op_end) {
		Pos = Op_end;
	}
	
	for (string com : commands) {
		if (com.compare("prev") == 0) {
			if (Pos - 10 > 0) {
				Pos -= 10;
			}
			else {
				Pos = 0;
			}
		}
		else if (com.compare("next") == 0) {
			if (Pos + 10 < Video_len) {
				Pos += 10;
			}
			else {
				Pos = Video_len;
			}
		}
		if (Op_start <= Pos && Pos <= Op_end) {
			Pos = Op_end;
		}

	}
	answer = (Pos / 60 < 10 ? "0" + to_string(Pos / 60) : to_string(Pos / 60))
		+ ":" + (Pos % 60 < 10 ? "0" + to_string(Pos % 60) : to_string(Pos % 60));




	return answer;
}